### PR TITLE
Add example of usage with Opella-XD debug probe

### PR DESCRIPTION
**Pull request by anthony-kolesov**
Signed-off-by: Anton Kolesov anton.kolesov@synopsys.com

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -213,6 +213,10 @@ will be printed on the server side.
 
 ### ARC EM application running on EM Starter Kit
 
+> Custom linker script is required to link applications for EM Starter Kit.
```
Comment (jeremybennett) [inline README.md:216]: Should read "A custom linker script..."

File: README.md [hunk 2]
```diff
@@ -213,6 +213,10 @@ will be printed on the server side.
...
+> Refer to section "Building application" of our EM Starter Kit Wiki page:
```
Comment (jeremybennett) [inline README.md:217]: Should read "Refer to the section..."

File: README.md [hunk 3]
```diff
@@ -213,6 +213,10 @@ will be printed on the server side.
...
+> https://github.com/foss-for-synopsys-dwc-arc-processors/toolchain/wiki/EM-Starter-Kit
+
 Download and build OpenOCD port for ARC as described here:
```
Comment (jeremybennett) [inline README.md:220]: Should read "Download and build the OpenOCD..."

File: README.md [hunk 4]
```diff
@@ -235,6 +239,52 @@ Compile and run:
     (gdb) break exit
     (gdb) continue
 
+
+### Debugging applications using Ashling Opella-XD debug probe
+
+> Custom linker script is required to link applications for EM Starter Kit.
```
Comment (jeremybennett) [inline README.md:245]: Should read "A custom linker script..."

File: README.md [hunk 5]
```diff
@@ -235,6 +239,52 @@ Compile and run:
...
+> Refer to section "Building application" of our EM Starter Kit Wiki page:
```
Comment (jeremybennett) [inline README.md:246]: Should read "Refer to the section..."

File: README.md [hunk 6]
```diff
@@ -235,6 +239,52 @@ Compile and run:
...
+> https://github.com/foss-for-synopsys-dwc-arc-processors/toolchain/wiki/EM-Starter-Kit
+> For different hardware configuration other changes might be required.
```
Comment (jeremybennett) [inline README.md:248]: Should read "For different hardware configurations other..."

File: README.md [hunk 7]
```diff
@@ -235,6 +239,52 @@ Compile and run:
...
+
+> Ashling Opella-XD debug probe and it's drivers are not part of GNU tools
```
Comment (jeremybennett) [inline README.md:250]: Should read "The Ashling Opella-XD debug probe and it's drivers are not part of the GNU tools"

File: README.md [hunk 8]
```diff
@@ -235,6 +239,52 @@ Compile and run:
...
+> distribution and should be obtained separately.
+
+Ashling Opella-XD drivers distribution contain gdbserver for GNU Tools. Start
```
Comment (jeremybennett) [inline README.md:253]: Should read "The Ashling Opella-XD..."

File: README.md [hunk 9]
```diff
@@ -235,6 +239,52 @@ Compile and run:
...
+it with following command:
+
+    $ ./ash-arc-gdb-server --jtag-frequency 8mhz --device arc \
+        --arc-reg-file <core.xml>
+
+Where <core.xml> is a path to XML file describing AUX registers of target core.
+Ashling drivers distribution contain files for ARC 600 (arc600-core.xml) and
```
Comment (jeremybennett) [inline README.md:260]: Should read "The Ashling drivers distribution contains files..."

File: README.md [hunk 10]
```diff
@@ -235,6 +239,52 @@ Compile and run:
...
+ARC 700 (arc700-core.xml). For EM additional download is required. Download
```
Comment (jeremybennett) [inline README.md:261]: Should read "... For EM an additional download..."

File: README.md [hunk 11]
```diff
@@ -235,6 +239,52 @@ Compile and run:
...
+arc-opella-em.xml file from here:
```
Comment (jeremybennett) [inline README.md:262]: you can omit the word "file"

File: README.md [hunk 12]
```diff
@@ -235,6 +239,52 @@ Compile and run:
...
+https://gist.github.com/anthony-kolesov/7193146.
+
+> Ashling gdbserver might emit error messages like "Error: Core is running".
```
Comment (jeremybennett) [inline README.md:265]: Should read "The Ashling gdbserver..."

File: README.md [hunk 13]
```diff
@@ -235,6 +239,52 @@ Compile and run:
...
+> Those messages are harmless and do not affect debugging experience.
```
Comment (jeremybennett) [inline README.md:266]: Should read "...affect the debugging experience."

File: README.md [hunk 14]
```diff
@@ -235,6 +239,52 @@ Compile and run:
...
+
+Then start GDB and connect to remote target as with OpenOCD/Linux gdbserver,
+etc. However unlike those target *before* connecting it is required to type
+`set arc opella-target arcem` command. Example for EM:
+
```
Comment (jeremybennett) [inline README.md:271]: I think the above three lines would be better reworked as:

Then start GDB. _Before_ connecting to an Opella-XD target it is essential to specify the architecture of the target. For EM use

  set arc opella-target arcem

(other possibilites are arc600 and arc700).

Then connect to the target as with the OpenOCD/Linux gdbserver. For example a full session with an Opella-XD controlling an ARC EM target could start as follows:

**Comment by jeremybennett** [comment]
Hi Anton,

Nice work. I've made a series of minor comments to polish the English, and one suggestion where I think the wording could be clearer.

Please commit and cherry-pick into the 4.8 branch as well.

**Comment by anthony-kolesov** [comment]
Updated and merged. I should reread rules on articles in English, never completly understood where to place them :)

**Comment by jeremybennett** [comment]
You should see what my Russian is like :)

**Pull request closed without merging**